components/motor/gpio/encoded.go

```diff
@@ -298,30 +281,32 @@ func (m *EncodedMotor) GoFor(ctx context.Context, rpm, revolutions float64, extr
 		10*time.Millisecond,
 		positionReached,
 	)
-	// Ignore the context canceled error - this occurs when the rpmCtx is canceled
-	// with m.rpmMonitorDone in goForInternal and in Stop
+	// Ignore the context canceled error - this occurs when the adjustmentsCtx is canceled
+	// with m.makeAdjustmentsDone in goForInternal and in Stop
 	if !errors.Is(err, context.Canceled) {
 		return err
 	}
 	return nil
 }
 
 func (m *EncodedMotor) goForInternal(rpm, goalPos, direction float64) error {
-	// cancel rpmMonitor if it already exists
-	if m.rpmMonitorDone != nil {
-		m.rpmMonitorDone()
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	// cancel makeAdjustments if it already exists
+	if m.makeAdjustmentsDone != nil {
+		m.makeAdjustmentsDone()
```

Comment:
are these something we usually care about locking for? it seems like we don't in other drivers, but I can't see why not?

Reply:
Depends on what's accessing the variable. Since this is populated in GoFor and is read from everywhere, I don't think there's a read/write race that we care about for offset, for example, which is suitable across API calls and multiple threads reading from it at the same time.  

This is probably a good addition, but we should work on mutex readability when we refactor. 

Mutexs aren't the only way to ensure thread safety too, you can look at the go examples for channels https://gobyexample.com/stateful-goroutines, or we could even store the simpler variables on the sturct in atomics. However, we'll keep the encoded motor's style because it's okay for this driver to rely on its mutexes; we'll just try to simplify the adjuster code further, and write more coverage tests for the math in the future.